lxd/instance/drivers/driver_qemu.go

```diff
@@ -2309,6 +2325,63 @@ func (d *qemu) deviceAttachBlockDevice(deviceName string, configCopy map[string]
 	return nil
 }
 
+func (d *qemu) unmountPath(deviceName string, rawConfig deviceConfig.Device) error {
+	// First, cleanly unmount the path inside the VM.
+	msg := map[string]any{
+		"action": "removed",
+		"name":   deviceName,
+		"config": rawConfig,
+	}
+
+	err := d.devlxdEventSend("device", msg)
+	if err != nil {
+		return err
+	}
+
+	// Now that the unmount has occurred on the VM,
```

Comment:
How do we know this? Perhaps a synchronous API event is what we need infact?

Or should the lxd-agent be responsible for cleaning up the mount and directories? 

Thoughts?

Reply:
@tomponline My thoughts on the second solution you suggests are: it seems easier and would require less changes, but I think there would be an issue in the end:

Since the `eventType` would be `unmount` as you asked above, I could pass a msg like:
```go
msg := map[string]any{
		"action": "removed",
		"name":   deviceName,
		"volatile": d.deviceVolatileGetFunc(deviceName)(),
		"config": rawConfig,
	}
``` 

and do the file remove logic inside the agent but in case where `last_state.created` is not present in the volatile key, I'll still need to do:

```go
err = d.deviceVolatileSetFunc(deviceName)(map[string]string{"last_state.created": ""})
if err != nil {
	return fmt.Errorf("Error updating volatile for the device: %w", err)
}
```

after the unmount occur successfully (but because this is not synchronous, when do I know when to do that ?)

Reply:
@tomponline other idea, specifically for the unmount event we could make the API synchronous and leave it async for the other actions:

```go
func eventsPost(d *Daemon, r *http.Request) response.Response {
	...

	// Handle device related actions locally.
	eventsProcess(event) // instead of `go eventsProcess(event)` 
	return response.SyncResponse(true, nil)
}

func eventsProcess(event api.Event) {
	...
	if e.Action == "unmount" {
	   ... // synchronous code
	   
	   return
	}
	
	// for other actions, async behaviour is used
	go func() {
	    if e.Action == "added" {
	       ...
	    }
	    
	    return
	}()
```

Reply:
fixed